### ISSUE TITLE
Can't cast a spell from spellbook

### ISSUE THREAD

**Issue by FrostSTeeL** [Hekili/hekili#502]
When i try to use a spell from my spell book (Pick Lock in this case) it gives me this error and won't let me cast the spell

```
build: v9.0.1-1.0.16
level: 50 (50)
class: ROGUE
spec: outlaw

talents: alacrity
    hit_and_run
    iron_stomach
    prey_on_the_weak
    quick_draw
    vigor

covenant: none

azerite: none

essences: [none], none

sets/legendaries/artifacts: none

gear: alderwood_loop = 1
    anglin_arts_treads = 1
    band_of_seething_manifest = 1
    banisher_armbands = 1
    banisher_waistcord = 1
    black_swashbucklers_shirt = 1
    crone_sticker = 1
    green_trophy_tabard_of_the_illidari = 1
    inherited_cape_of_the_black_baron = 1
    kralzas_resolve = 1
    mr_munchykins = 1
    neglected_luck_pendant = 1
    overgrowth_cutter_gloves = 1
    stained_shadowcraft_cap = 1
    stained_shadowcraft_pants = 1
    stained_shadowcraft_spaulders = 1
    stained_shadowcraft_tunic = 1
    venerable_dalrends_sacred_charge = 1

legendaries: none

itemIDs: 112595, 113162, 122248, 122254, 122261, 122351, 122358, 122383, 155158, 155468, 155474, 155567, 155584, 158598, 158616, 179385, 31404, 4336

settings: aoe = 3
    buffPadding = 0
    custom1Name = Custom 1
    custom2Name = Custom 2
    cycle = false
    cycle_min = 6
    damage = true
    damageDots = false
    damageExpiration = 6
    damagePets = false
    damageRange = 0
    debuffPadding = 0
    enabled = true
    gcdSync = true
    maxRefresh = 10
    maxTime = 33
    nameplateRange = 8
    nameplates = true
    package = Outlaw
    petbased = false
    potion = potion_of_unbridled_fury
    potionsReset = 20180919.1
    throttleRefresh = false
    throttleTime = false
    mfd_waste = true

toggles: cooldowns = true
    custom1 = false
    custom2 = false
    defensives = false
    essences = true
    interrupts = false
    mode = automatic
    potions = false
```
```
Message: ADDON_ACTION_FORBIDDEN: Hekili tried to call the protected function 'CastSpell()'.
Time: Mon Oct 26 05:39:45 2020
Count: 5
Stack: ADDON_ACTION_FORBIDDEN: Hekili tried to call the protected function 'CastSpell()'.
[string "@Interface\AddOns\ElvUI\Libraries\Ace3\CallbackHandler-1.0\CallbackHandler-1.0.lua"]:29: in function <...ies\Ace3\CallbackHandler-1.0\CallbackHandler-1.0.lua:25>
[string "@Interface\AddOns\ElvUI\Libraries\Ace3\CallbackHandler-1.0\CallbackHandler-1.0.lua"]:64: in function `Fire'
[string "@Interface\AddOns\ElvUI\Libraries\Ace3\AceEvent-3.0\AceEvent-3.0.lua"]:120: in function <...s\ElvUI\Libraries\Ace3\AceEvent-3.0\AceEvent-3.0.lua:119>
[string "=[C]"]: in function `CastSpell'
[string "@Interface\FrameXML\SpellBookFrame.lua"]:639: in function `SpellButton_OnClick'
[string "*:OnClick"]:4: in function <[string "*:OnClick"]:1>

Locals: <none>

```
_Edit: Formatting_

**Comment by penguinsane** [comment]
I have the same issue, but here is my error:

```
2x [ADDON_ACTION_FORBIDDEN] AddOn 'Hekili' tried to call the protected function 'CastSpell()'.
[string "@!BugGrabber\BugGrabber.lua"]:519: in function <!BugGrabber\BugGrabber.lua:519>
[string "=[C]"]: in function `CastSpell'
[string "@FrameXML\SpellBookFrame.lua"]:639: in function `SpellButton_OnClick'
[string "*:OnClick"]:4: in function <[string "*:OnClick"]:1>
```

**Comment by Hekili** [comment]
Can you replicate this with all other addons except BugSack/BugGrabber disabled?

**Comment by FrostSTeeL** [comment]
I don't have BugSack/BugGrabber but it sill happens

**Comment by Crelam** [comment]
You misunderstand.  Hekili is asking you to disable all addons except Hekili, Buggrabber, & Bugsack.  If you don't have Buggrabber & Bugsack, please install them.  These addons capture error messages.

https://www.curseforge.com/wow/addons/bug-grabber
https://www.curseforge.com/wow/addons/bugsack

**Comment by FrostSTeeL** [comment]
Ok sorry i misunderstood. I installed the BugGrabber and BugSack.

**When i have everything enabled i get this;**

```
1x [ADDON_ACTION_FORBIDDEN] AddOn 'Hekili' tried to call the protected function 'CastSpell()'.
[string "@!BugGrabber\BugGrabber.lua"]:519: in function <!BugGrabber\BugGrabber.lua:519>
[string "=[C]"]: in function `CastSpell'
[string "@FrameXML\SpellBookFrame.lua"]:639: in function `SpellButton_OnClick'
[string "*:OnClick"]:4: in function <[string "*:OnClick"]:1>
```

**When i disable BugGrabber and BugSack but keep everything else**, it actually works, it didn't work yesterday and i don't think ElvUI got an update.

**When i disable everything but BugGrabber and BugSack,** No bugs either

I tried everything again, but now it seems everything is working. It went away by itself i guess.


```
build: v9.0.1-1.0.16
level: 50 (50)
class: ROGUE
spec: outlaw

talents: alacrity
    hit_and_run
    iron_stomach
    prey_on_the_weak
    quick_draw
    vigor

covenant: none

azerite: none

essences: [none], none

sets/legendaries/artifacts: none

gear: alderwood_loop = 1
    anglin_arts_treads = 1
    band_of_seething_manifest = 1
    banisher_armbands = 1
    banisher_waistcord = 1
    black_swashbucklers_shirt = 1
    crone_sticker = 1
    green_trophy_tabard_of_the_illidari = 1
    inherited_cape_of_the_black_baron = 1
    kralzas_resolve = 1
    mr_munchykins = 1
    neglected_luck_pendant = 1
    overgrowth_cutter_gloves = 1
    stained_shadowcraft_cap = 1
    stained_shadowcraft_pants = 1
    stained_shadowcraft_spaulders = 1
    stained_shadowcraft_tunic = 1
    venerable_dalrends_sacred_charge = 1

legendaries: none

itemIDs: 112595, 113162, 122248, 122254, 122261, 122351, 122358, 122383, 155158, 155468, 155474, 155567, 155584, 158598, 158616, 179385, 31404, 4336

settings: aoe = 3
    buffPadding = 0
    custom1Name = Custom 1
    custom2Name = Custom 2
    cycle = false
    cycle_min = 6
    damage = true
    damageDots = false
    damageExpiration = 6
    damagePets = false
    damageRange = 0
    debuffPadding = 0
    enabled = true
    gcdSync = true
    maxRefresh = 10
    maxTime = 33
    nameplateRange = 8
    nameplates = true
    package = Outlaw
    petbased = false
    potion = potion_of_unbridled_fury
    potionsReset = 20180919.1
    throttleRefresh = false
    throttleTime = false
    mfd_waste = true

toggles: cooldowns = true
    custom1 = false
    custom2 = false
    defensives = false
    essences = true
    interrupts = false
    mode = automatic
    potions = false
```

**Comment by Hekili** [comment]
Sorry, to be more clear:  I'm looking at whether, with only Hekili + BugSack + BugGrabber, you get the error.

This looks like a taint error, and a problem with taint errors is that they falsely attribute blame to the wrong source.  My addon literally does not touch the spellbook or its clickable elements in any way.

**Comment by penguinsane** [comment]
I get the same results as FrostSTeeL with just Hekili, Bugsack, and Buggrabber. Sometimes it resolves itself, sometimes it doesn't. I understand that it may not be specifically an issue with your addon, but this is an issue that only recently popped up in the last few versions of Hekili, when previously both Hekili and Bugsack/grabber worked fine together. I am not sure which author would need to be the one that fixes it (or if it's fixable by either, idk how taint errors really work), so I was not sure where to report it.

**Comment by Hekili** [comment]
It's not an interaction between BugSack/BugGrabber and Hekili.  BugSack/BugGrabber are just addons that make it easy to catch error messages.  Please don't assume that I'm saying something I'm not.

What I am saying is, the situation that causes this error isn't something that the addon does.  It might not even be another addon -- it might be a bug in the default UI.  In order to find out, I need someone to actually answer my question:  Does it happen with _only_ Hekili loaded, along with BugSack/BugGrabber (to catch the error messages)?  That is, Hekili + BugSack/BugGrabber + *no other addons*.

**Comment by FrostSTeeL** [comment]
As i said above, i tried it in total of 6 times in these combinations, 2 times all the addons enabled, 2 times just hekili and bugsack+buggrabber, 2 times all addons enabled without bugsack and buggrabber. In those 6 tires, sometimes it gives the error, sometimes it does not no matter what the addon combinations are. Are there anything else i can do to help you ?

**Comment by Hekili** [comment]
What you said there doesn't match what you said above, but that's okay.  I'll test a fix and we'll see what happens from there.  Thanks.

**Comment by Hekili** [comment]
I've been investigating this and will continue to investigate.  It's really annoying.  I'm not touching anything I'm not supposed to touch, but here we are.

**Comment by Hekili** [comment]
This was a pain in the ass to solve, but I think it's fixed in 1.0.17-beta5.

**Comment by FrostSTeeL** [comment]
Thank you for your hard work! You are awesome!
